examples/structured_2d_dgsem/elixir_mhd_coupled.jl

```diff
@@ -0,0 +1,135 @@
+                       coordinates_min1,
+                       coordinates_max1,
+                       periodicity = (false, true))
+
+coupling_function1 = (x, u, equations_other, equations_own) -> u
+boundary_conditions1 = (x_neg = BoundaryConditionCoupled(2, (:end, :i_forward), Float64,
+                                                         coupling_function1),
+                        x_pos = BoundaryConditionCoupled(2, (:begin, :i_forward), Float64,
+                                                         coupling_function1),
+                        y_neg = boundary_condition_periodic,
+                        y_pos = boundary_condition_periodic)
+
+semi1 = SemidiscretizationHyperbolic(mesh1, equations, initial_condition1, solver,
+                                     boundary_conditions = boundary_conditions1)
+
+###########
+# system #2
+###########
+
+initial_condition2 = initial_condition_convergence_test
+coordinates_min2 = (0.0, -1 / sin(pi / 4))
+coordinates_max2 = (1 / sin(pi / 4), 1 / sin(pi / 4))
+mesh2 = StructuredMesh(cells_per_dimension,
+                       coordinates_min2,
+                       coordinates_max2,
+                       periodicity = (false, true))
+
+coupling_function2 = (x, u, equations_other, equations_own) -> u
+boundary_conditions2 = (x_neg = BoundaryConditionCoupled(1, (:end, :i_forward), Float64,
+                                                         coupling_function2),
+                        x_pos = BoundaryConditionCoupled(1, (:begin, :i_forward), Float64,
+                                                         coupling_function2),
+                        y_neg = boundary_condition_periodic,
+                        y_pos = boundary_condition_periodic)
+
+semi2 = SemidiscretizationHyperbolic(mesh2, equations, initial_condition2, solver,
+                                     boundary_conditions = boundary_conditions2)
+
+# Create a semidiscretization that bundles all the semidiscretizations.
+semi = SemidiscretizationCoupled(semi1, semi2)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 0.1)
+ode = semidiscretize(semi, tspan)
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 100
+
+analysis_callback1 = AnalysisCallback(semi1, interval = 100)
+analysis_callback2 = AnalysisCallback(semi2, interval = 100)
+analysis_callback = AnalysisCallbackCoupled(semi, analysis_callback1, analysis_callback2)
+
+alive_callback = AliveCallback(analysis_interval = analysis_interval)
+
+save_solution = SaveSolutionCallback(interval = 50,
+                                     save_initial_solution = true,
+                                     save_final_solution = true,
+                                     solution_variables = cons2prim)
+
+cfl = 1.0
+
+stepsize_callback = StepsizeCallback(cfl = cfl)
+
+glm_speed_callback = GlmSpeedCallback(glm_scale = 0.5, cfl = cfl, semi_indices = tuple(1))
```

Comment:
A final question. Because the default of the keyword argument `semi_indices` is `()`, what happens if one attempts to run this elixir with standard inputs? I.e, the callback constructor is
```julia
glm_speed_callback = GlmSpeedCallback(glm_scale = 0.5, cfl = cfl)
```
Does the simulation erroneously run? Or does an error get thrown?

Reply:
We get an error message in that case.

Reply:
How does it look like? Is it an informative one that makes clear that one need to overwrite the default parameter in the case of a coupled semidiscretizations?

Reply:
Good point. We now throw an error if that is not the case:
`ERROR: LoadError: "Equation of semidiscretization 2 needs to be included in 'semi_indices' of 'GlmSpeedCallback'."`

Reply:
We are now getting a descriptive error message:
`ERROR: LoadError: "Equation of semidiscretization 2 needs to be included in 'semi_indices' of 'GlmSpeedCallback'."`

Also, if no `semi_indices` are specified we now get this error message:
`ERROR: LoadError: "Since you have more than one semidiscretization you need to specify the 'semi_indices' for which the GLM speed needs to be calculated."`

Reply:
@andrewwinters5000 @JoshuaLampert Has the raised issue been resolved from your POV?

Reply:
Yes :+1: